src/components/DecisioningEngine/index.js

```diff
@@ -48,6 +48,26 @@ const createDecisioningEngine = ({ config, createNamespacedStorage }) => {
       },
       onComponentsRegistered(tools) {
         applyResponse = createApplyResponse(tools.lifecycle);
+        if (personalizationStorageEnabled) {
+          const storage = createNamespacedStorage(
+            `${sanitizeOrgIdForCookieName(orgId)}.decisioning.`
+          );
+          consent
+            .awaitConsent()
+            .then(() => {
+              const eventRegistry = createEventRegistry({
+                storage: storage.persistent
+              });
+              eventContext.setCurrentEventRegistry(eventRegistry);
+            })
+            .catch(() => {
+              // If consent is rejected, we want to clear the local storage.
+              if (storage) {
+                storage.persistent.clear();
```

Comment:
If a customer sets `personalizationStorageEnabled` to false at some point, local storage should also be cleared.  That way customer's won't disable it and wonder why they still see an entry in local storage.